packages/audioplayers/tizen/src/audio_player.cc

```diff
@@ -42,164 +39,148 @@ void AudioPlayer::Play() {
+        HandleResult("player_set_uri", ret);
       }
       should_play_ = true;
       PreparePlayer();
       break;
     case PLAYER_STATE_READY:
     case PLAYER_STATE_PAUSED:
-      result = player_start(player_);
-      HandleResult("player_start", result);
+      ret = player_start(player_);
+      HandleResult("player_start", ret);
       should_play_ = false;
-      start_playing_listener_(player_id_);
+      EmitPositionUpdates();
       break;
     default:
-      LOG_DEBUG("player is already playing audio");
+      // Player is already playing audio.
       break;
   }
 }
 
 void AudioPlayer::Pause() {
-  LOG_INFO("AudioPlayer %s is pausing...", player_id_.c_str());
   if (GetPlayerState() == PLAYER_STATE_PLAYING) {
-    int result = player_pause(player_);
-    HandleResult("player_pause", result);
+    int ret = player_pause(player_);
+    HandleResult("player_pause", ret);
   }
   should_play_ = false;
 }
 
 void AudioPlayer::Stop() {
-  LOG_INFO("AudioPlayer %s is stopping...", player_id_.c_str());
-  if (release_mode_ == RELEASE) {
+  if (release_mode_ == ReleaseMode::kRelease) {
     Release();
   } else {
     player_state_e state = GetPlayerState();
     if (state == PLAYER_STATE_PLAYING || state == PLAYER_STATE_PAUSED) {
-      int result = player_stop(player_);
-      HandleResult("player_stop", result);
+      int ret = player_stop(player_);
+      HandleResult("player_stop", ret);
     }
   }
   should_play_ = false;
   seeking_ = false;
 }
 
 void AudioPlayer::Release() {
-  LOG_INFO("AudioPlayer %s is releasing...", player_id_.c_str());
-  if (player_ != nullptr) {
+  if (player_) {
     player_unset_completed_cb(player_);
     player_unset_interrupted_cb(player_);
     player_unset_error_cb(player_);
     player_destroy(player_);
     player_ = nullptr;
   }
+  if (timer_) {
+    ecore_timer_del(timer_);
+    timer_ = nullptr;
+  }
 }
 
 void AudioPlayer::Seek(int position) {
-  LOG_INFO("AudioPlayer %s is seeking...", player_id_.c_str());
   if (seeking_) {
-    LOG_DEBUG("player is already seeking, can't seek again");
     return;
   }
 
   player_state_e state = GetPlayerState();
   if (state == PLAYER_STATE_READY || state == PLAYER_STATE_PLAYING ||
       state == PLAYER_STATE_PAUSED) {
-    LOG_DEBUG("set play position %d", position);
     seeking_ = true;
-    int result = player_set_play_position(player_, position, true,
-                                          OnSeekCompleted, (void *)this);
-    if (result != PLAYER_ERROR_NONE) {
+    int ret = player_set_play_position(player_, position, true, OnSeekCompleted,
+                                       (void *)this);
+    if (ret != PLAYER_ERROR_NONE) {
       seeking_ = false;
-      std::string error(get_error_message(result));
+      std::string error(get_error_message(ret));
       LOG_ERROR("player_set_play_position failed : %s", error.c_str());
       throw AudioPlayerError(error, "player_set_play_position failed");
```

Comment:
I assumed that original code didn't use `HandleResult` here because of the statement `seeking_ = false`. The fixed code removes `HandleResult` completely.

Reply:
Setting the `seeking_` value after calling `player_set_play_position` seems unsafe. It might lead to a race condition or even `OnSeekCompleted` might be called synchronously before `player_set_play_position` returns.

Reply:
Wow you are right, I'll fix it back.